tests/playwright/resources/create-extension.sh

```diff
@@ -0,0 +1,36 @@
+: '/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/'
+
+#!/bin/bash
+
+extensionFolder=bootcextensionlocal
+
+yarn build
+mkdir -p $extensionFolder
+cp -r packages/backend/dist/ $extensionFolder/dist
+cp packages/backend/package.json $extensionFolder/
+cp -r packages/backend/media/ $extensionFolder/media
+cp LICENSE $extensionFolder/
+cp packages/backend/icon-dark.png $extensionFolder/
+cp packages/backend/icon-light.png $extensionFolder/
+cp packages/backend/bootable.woff2 $extensionFolder/
+cp README.md $extensionFolder/
```

Comment:
I don't really know what it does but we should not do this because it'll lead to errors, unmaintained stuff, etc.

The extension has a Containerfile that builds the extension.

if we need to have a 'build-ready' folder of the extension, it should reuse that Containerfile has it's the own that will package the extension.

Else for example you may have omitted here some files that are included correctly in the release or you may have files that are there but not in the Containerfile so PR will pass but end users will not have the extension working

to me e2e tests are to really tests what is being 'produced' so it should reuse the build part of the extension

Reply:
@benoitf The script represents how to build the extension locally, given that it's supposed to run on each individual branch, if you have another solution of how that can be achieved that would be great.

Reply:
I agree. 
Although, having extension built locally (and tagged) using containerfile will not help us either, since then I have problem with authentication of the local repository when using locally built image as an input for install extension from OCI image. So we do not have a valid image from upstream on pr check (we might consider adding that, but then we are facing problem with secrets... )

Do you suggest something like this then https://github.com/odockal/podman-desktop-qe/blob/main/scripts/extension-installation-local.sh#L37-L53 ?

Otherwise I do not know how to do it.

Reply:
so I still believe we should try to install the image locally built to really test what we're building and how users will handle the image
but we can 'postpone' the issue you seem to have with authentication

as intermediate step, yes let's copy the content of the image

1. build the image
2. create the container like `CONTAINER_ID=$(podman create my-local-bootc-image)`
3. dump the content `podman export $CONTAINER_ID > /tmp/content.tar`
4. extract/save the content to `~/.local/share/containers/podman-desktop/plugins` folder (as export will have an extension folder, we will have ``~/.local/share/containers/podman-desktop/plugins/extension` once extracted

and let's replace soon that by installing the extension from Podman Desktop itself

Reply:
and no need to use `--extension-folder` arg when launching Podman Desktop

Reply:
also we should launch Podman Desktop binary (the one of production mode) not in development mode but I think here it's a 'known issue' about being able to run electron + playwright without dev mode

but at least we should be testing the 'production bits' of the extension, not the development mode

Reply:
@benoitf @cbr7 That sounds doable then! Thanks!

Reply:
also could you say what is the issue if you publish the image to a local registry 

AFAIK this one is already fixed: https://github.com/containers/podman-desktop/issues/6344

Reply:
as I guess it would be better if we could publish to localhost:500 registry and install the extension from that registry as part of the workflow

Reply:
@benoitf the error says:
```
Error while analyzing image: Error: Unable to find auth info for https://localhost/v2/. Error: RequestError
```

which might imply that the image itself is not part of any registry? 
I am simply building/tagging the image locally. no push into local registry. That might be a requirement?

Reply:
The extension image is pulled by Podman Desktop itself (not by podman). localhost for Podman Desktop means your laptop, not the vm where podman machine might run. And it's discussing through registry protocol, so it'll call `/v2` path on the path to interact with.

so yes, if you want to access something being built by Podman, image needs to be exposed to a registry first.

Reply:
@benoitf oh, alright, thanks for explanation. I am trying to set it up locally, so we could automate in the workflow, but I am not really successful atm (I tried what you are proposing in the #6344, though I got `panic: unable to configure authorization (htpasswd): open /auth/htpasswd: permission denied`). I would propose that we do now `build+export+extract` path how to install the extension, and then create a follow up issue for `building + pushing + installing` the extension via oci image from local registry, which gonna require some more testing. wdyt?

Reply:
sure one problem after another one

most important one is to use the artifact being produced ( consuming it from the OCI registry is another step)

Reply:
Perfect!

Reply:
Should be good now, the extension is built from the container file and extracted into the plugin folder to be picked up when PD starts up.

Reply:
Follow up issue: https://github.com/containers/podman-desktop-extension-bootc/issues/330